pkg/core/state/deposit.go

```diff
@@ -15,12 +18,46 @@ type Deposit struct {
 
 // EncodeBinary implements io.Serializable interface.
 func (d *Deposit) EncodeBinary(w *io.BinWriter) {
-	w.WriteVarBytes(bigint.ToBytes(d.Amount))
-	w.WriteU32LE(d.Till)
+	stackitem.EncodeBinary(d.toStackItem(), w)
 }
 
 // DecodeBinary implements io.Serializable interface.
 func (d *Deposit) DecodeBinary(r *io.BinReader) {
-	d.Amount = bigint.FromBytes(r.ReadVarBytes())
-	d.Till = r.ReadU32LE()
+	si := stackitem.DecodeBinary(r)
+	if r.Err != nil {
+		return
+	}
+	r.Err = d.fromStackItem(si)
+}
+
+func (d *Deposit) toStackItem() stackitem.Item {
+	return stackitem.NewStruct([]stackitem.Item{
+		stackitem.NewBigInteger(d.Amount),
+		stackitem.Make(d.Till),
+	})
+}
+
+func (d *Deposit) fromStackItem(it stackitem.Item) error {
+	items, ok := it.Value().([]stackitem.Item)
+	if !ok {
+		return errors.New("not a struct")
+	}
+	if len(items) != 2 {
+		return errors.New("wrong number of elements")
+	}
+	amount, err := items[0].TryInteger()
+	if err != nil {
+		return fmt.Errorf("invalid amount: %w", err)
+	}
+	till, err := items[1].TryInteger()
+	if err != nil {
+		return fmt.Errorf("invalid till: %w", err)
+	}
+	ti64 := till.Int64()
+	if !till.IsInt64() || ti64 > math.MaxUint32 || ti64 < 0 {
```

Comment:
We can do `IsUint64` and omit `< 0` comparison. We convert it later anyway.

Reply:
We're extracting with `Int64()`, so that's the one to check with `Is...`.

Reply:
I mean what prevents us to extract `Uint64()`?

Reply:
Fixed.